### ISSUE TITLE
Null Widget Received

### ISSUE THREAD

**Issue by colbyharrison** [Workday/autoparse-json#3]
The parser throws an exception when receiving a response that contains a null widget. 
`{
    "widget":null
}`

The exception occurs in JsonParserUtils.java, line 541. Looks like we assume reader will have a nextString() if `discriminationKeyName.equals(firstName)`, is true. 

![image](https://cloud.githubusercontent.com/assets/11632011/19253432/de098f44-8eff-11e6-9c94-9ef6c7a12691.png)

Is this the expected behavior? It might be more useful to ignore the null case, and move on to the next object. 

For cases such as these, should it be the responsibility of the parser to gracefully deal with null cases, or the responsibility of the server to send valid objects?

**Comment by ndtaylor** [comment]
Yes, an exception being thrown is the expected behavior, because the widget field being null would never have happened in our case. But, I think it would be perfectly valid to change the logic to fall back to the UnknownObjectParser or return a JSONObject if the widget field is null.
